src/utils/daos_hdlr.c

```diff
@@ -2643,42 +2636,63 @@ dm_connect(struct cmd_args_s *ap,
 					      "of the form:\n\t --dst </$pool> | </path/to/uns>");
 				D_GOTO(err, rc);
 			}
-			uuid_copy(ca->dst_c_uuid, dattr.da_cuuid);
+			snprintf(ca->dst_cont, DAOS_PROP_LABEL_MAX_LEN + 1, "%s", dattr.da_cont);
 		}
-		/* try to open container if this is a filesystem copy,
-		 * and if it fails try to create a destination,
-		 * then attempt to open again
+
+		/* try to open container if this is a filesystem copy, and if it fails try to create
+		 * a destination, then attempt to open again
 		 */
-		rc = daos_cont_open(ca->dst_poh, ca->dst_c_uuid, DAOS_COO_RW, &ca->dst_coh,
-				    dst_cont_info, NULL);
+		if (dst_cont_passed)
+			rc = daos_cont_open(ca->dst_poh, ca->dst_cont, DAOS_COO_RW, &ca->dst_coh,
+					    dst_cont_info, NULL);
+		else
+			rc = -DER_NONEXIST;
 		if (rc == -DER_NONEXIST) {
+			uuid_t cuuid;
+
 			if (ca->cont_layout == DAOS_PROP_CO_LAYOUT_POSIX) {
-				rc = dfs_cont_create(ca->dst_poh, ca->dst_c_uuid, NULL, NULL, NULL);
+				if (dst_cont_passed) {
```

Comment:
@mchaarawi I am just curious so dfs_cont_create requires a uuid_t (there's no label version) but daos_cont_create has a label and uuid version?

Reply:
it doesn't require a uuid. it lets you pass one or return one for you.
you can set the label with the property if you want it.
a new dfs_cont_create_with_label is being added here: https://github.com/daos-stack/daos/pull/6488/

Reply:
This won't yet work if a user specifies a container label like this, right?
```
daos://my_pool/my_cont/my_dir
```

Reply:
@mchaarawi ok I guess I meant you pass in uuid to dfs_cont_create or it passes back uuid, but you answered my question about a dfs_cont_create_with_label version.

Reply:
right.. we need a different PR to handle creating the container with a label.
probable after that other PR lands.

Reply:
if the container exists though, i think this should work.

Reply:
had to fix a conflict.
Danielle i imported your change to increase the timeout too here, so maybe can close your PR to not generate another conflict?